ray-curator/ray_curator/stages/text/deduplication/removal_workflow.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Literal, Optional
+
+from loguru import logger
+
+from ray_curator.stages.base import ProcessingStage
+from ray_curator.stages.deduplication.id_generator import CURATOR_DEDUP_ID_STR
+from ray_curator.tasks import FileGroupTask
+
+from .removal import TextDuplicatesRemovalStage
+
+if TYPE_CHECKING:
+    from ray_curator.backends.base import BaseExecutor
+
+
+@dataclass
+class TextDuplicatesRemovalWorkflow:
+    # required args
+    input_path: str
```

Comment:
nit: can be optional if `initial_tasks` is not None

Reply:
fixed, and added a test